pallets/liquidity-pools/src/inbound.rs

```diff
@@ -305,44 +303,13 @@ where
 		tranche_id: T::TrancheId,
 		investor: T::AccountId,
 		currency_index: GeneralCurrencyIndexOf<T>,
-		destination: DomainAddress,
 	) -> DispatchResult {
 		let invest_id: T::TrancheCurrency = Self::derive_invest_id(pool_id, tranche_id)?;
-		let currency_index_u128 = currency_index.index;
 		let payout_currency = Self::try_get_payout_currency(invest_id.clone(), currency_index)?;
```

Comment:
**CR**

Looking at the code of `try_get_payout_currency` I think this is not working. 

The check that is being used `T::ForeignInvestment::accepted_payment_currency(invest_id,currency)` will check the wrong trading pair. I.e. the wrong trading pair direction. As trading pairs are not ensured to work in both ways, this might return ok also it will fail when trying to swap.

Reply:
Great catch! Will fix `fn try_get_payout_currency`.

Reply:
I can not, but it might still happen by accident as no logic in the order book ensures bidirectionality/

Reply:
On that note: If bidirectionality is not given, an investor cannot decrease their foreign investment if it creates an orderbook swap from `pool` to `foreign` because then `TradingPair::<T>::get(pool_currency, foreign_currency)`  does not exist. Moreover, we block redemptions for that `foreign` currency as well.

AFAICT, we should check in `fn allow_investment_currency` both if the currency is accepted as payment as well as payout, not only the former. WDYT?

Reply:
> AFAICT, we should check in fn allow_investment_currency both if the currency is accepted as payment as well as payout, not only the former. WDYT?

cc @hieronx

Reply:
Yes 100%. I think in practice governance should ensure that always both directions of an LP token trading pair are added.